tests/snippets/invalid_syntax.py

```diff
@@ -0,0 +1,16 @@
+
+
+src = """
+def valid_func():
+    pass
+
+yield 2
+"""
+
+try:
+    compile(src, 'test.py', 'exec')
+except SyntaxError as ex:
+    assert ex.lineno == 5
+else:
+    raise AssertionError("Must throw syntax error")
```

Comment:
You can use `assertRaises` here

Reply:
I also want to check the `lineno` attribute, so thats why I did not use it.